clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -67,8 +69,21 @@
   @Getter
   private final List<String> oAuthScopes;
 
-  @Getter
   private final String defaultNamespace = "default";
+  private final Path serviceAccountNamespacePath = Paths.get("/var/run/secrets/kubernetes.io/serviceaccount/namespace");
+  public String getDefaultNamespace() {
+    String namespace = defaultNamespace;
+    try {
+      Optional<String> serviceAccountNamespace = Files.lines(serviceAccountNamespacePath, StandardCharsets.UTF_8).findFirst();
+      namespace = serviceAccountNamespace.get();
+    } catch (IOException | NoSuchElementException e) {
+      try {
+        namespace = jobExecutor.defaultNamespace(this);
```

Comment:
What happens when this is empty? (Seems like it can be [here](https://github.com/spinnaker/clouddriver/pull/2279/files#diff-65ee14f962603a2d9dccefbbc331e68cR94))

Reply:
Yes, good catch. I refactored this a bit too quickly. Fixed.